linter.lua

```diff
@@ -25,10 +26,53 @@ local function run_lint_cmd(path, linter)
 end
 
 
+local function match_pattern(text, pattern, order)
+  if order == nil then
+    return text:gmatch(pattern)
+  end
+
+  return coroutine.wrap(function()
+    for one, two, three in text:gmatch(pattern) do
+      local line, col, warn
+      for field,position in pairs(order) do
+        local value
+        if position == 1 then
+          value = one
+        elseif position == 2 then
+          value = two
+        elseif position == 3 then
+          value = three
+        end
+
+        if field == "line" then
```

Comment:
Instead of a switch for each element and checking for the default of each value a table with the values already ordered by the wanted output could be used. Also when declaring the table the default values can be already there.
Add the ordered_table outside of the loop
```lua
local ordered_fields = {1, 1, "syntax error"}
```
Set the values according to the position set in the configuration
```lua
if value then
  ordered_fields[position] = value
  if field ~= "message" and current_doc ~= nil and tonumber(value) > #current_doc.lines then
    ordered_fields[position] = #current_doc.lines
  end
end
```